### PR TITLE
improve: change the default status bar overlay color in dark mode (SDKCF-3303)

### PR DESCRIPTION
**Pull request by imaginaris**
## Links
SDKCF-3303

# Checklist
- [X] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [ ] I wrote/updated tests for new/changed code
- [X] I removed all sensitive data **before every commit**, including API endpoints and keys
- [ ] I ran `fastlane ci` without errors

### CODE REVIEW

File: RInAppMessaging/Classes/Extensions/UIApplication+IAM.swift [hunk 1]
```diff
@@ -21,7 +21,15 @@ extension UIApplication {
                 .filter({ $0.activationState == .foregroundActive })
                 .compactMap({ $0 as? UIWindowScene })
                 .first {
-            return keyScene.statusBarManager?.statusBarStyle
+
+            var style = keyScene.statusBarManager?.statusBarStyle
+            // `default` style doesn't tell us anything, so we can fallback to the most common pattern:
+            // light status bar content in dark mode and dark content in light mode.
+            if style == .default, let keyWindow = getKeyWindow() {
+                let isDarkMode = keyWindow.traitCollection.userInterfaceStyle == .dark
```
Comment (donnie-jp) [inline RInAppMessaging/Classes/Extensions/UIApplication+IAM.swift:29]: It seems like `userInterfaceStyle` is only available >= iOS 12?
https://developer.apple.com/documentation/uikit/uitraitcollection/1651063-userinterfacestyle

Comment (imaginaris) [reply RInAppMessaging/Classes/Extensions/UIApplication+IAM.swift:29]: True, but `.darkContent` isn't (status bar style). I would have to use my own enum to support that piece of logic on iOS 12. Do you think it's worth it?

Comment (donnie-jp) [reply RInAppMessaging/Classes/Extensions/UIApplication+IAM.swift:29]: From the diff context I couldn't see that this was already wrapped in an `#available(iOS 13.0, *)` check. I was worried about a runtime crash. I tried putting the `let isDarkMode` line into an app and it wouldn't even compile without an availability check - Swift FTW! 😅

**Review by donnie-jp — APPROVED** [review]
LGTM

**Pull request merged**